src/branch.rs

```diff
@@ -28,6 +28,18 @@ impl<'repo> Branch<'repo> {
         Branch { inner: reference }
     }
 
+    /// Ensure the branch name is well-formed.
+    pub fn name_is_valid(name: &str) -> bool {
+        let name = CString::new(name).unwrap();
+        unsafe {
+            let mut valid: libc::c_int = 0;
+            // TODO:check that git_branch_name_is_valid returns 0?
+            // what should we do if it doesn't?
```

Comment:
check out `delete` and the use of `try_call` to see how to handle the return values. this allows us to return a `Result`

Reply:
thanks for the pointers! returning `Result<bool>` now.